core/amber/src/test/scala/edu/uci/ics/texera/workflow/operators/specializedFilter/SpecializedFilterOpExecSpec.scala

```diff
@@ -0,0 +1,226 @@
+import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, AttributeType, Schema}
+import edu.uci.ics.texera.workflow.operators.filter.{
+  ComparisonType,
+  FilterPredicate,
+  SpecializedFilterOpDesc,
+  SpecializedFilterOpExec
+}
+import org.scalatest.BeforeAndAfter
+import org.scalatest.flatspec.AnyFlatSpec
+
+import java.util.Arrays.asList
+
+class SpecializedFilterOpExecSpec extends AnyFlatSpec with BeforeAndAfter {
+  var counter: Int = 0
+
+  val tupleSchema: Schema = Schema
+    .newBuilder()
+    .add(new Attribute("string", AttributeType.STRING))
+    .add(new Attribute("int", AttributeType.INTEGER))
+    .add(new Attribute("bool", AttributeType.BOOLEAN))
+    .add(new Attribute("long", AttributeType.LONG))
+    .build()
+
+  def layerID(): LayerIdentity = {
+    counter += 1
+    LayerIdentity("" + counter, "" + counter, "" + counter)
+  }
+
+  def linkID(): LinkIdentity = LinkIdentity(layerID(), layerID())
+
+  def markerValue(): String = {
+    "RandomValue"
+  }
+
+  def allNullTuple(): Tuple = {
+    counter += 1
+    Tuple
+      .newBuilder(tupleSchema)
+      .add(new Attribute("string", AttributeType.STRING), null)
+      .add(new Attribute("int", AttributeType.INTEGER), null)
+      .add(new Attribute("bool", AttributeType.BOOLEAN), null)
+      .add(new Attribute("long", AttributeType.LONG), null)
+      .build()
+  }
+
+  def nonNullTuple(): Tuple = {
+    counter += 1
+    Tuple
+      .newBuilder(tupleSchema)
+      .add(new Attribute("string", AttributeType.STRING), "hello")
+      .add(new Attribute("int", AttributeType.INTEGER), 0)
+      .add(new Attribute("bool", AttributeType.BOOLEAN), false)
+      .add(new Attribute("long", AttributeType.LONG), Long.MaxValue)
+      .build()
+  }
+
+  before {
+    counter = 0
+  }
+
+  it should "open and close" in {
+    val opExec = new SpecializedFilterOpExec(new SpecializedFilterOpDesc())
+    opExec.open()
+    opExec.close()
+  }
+
+  it should "include all tuples from 1 completely null input stream of string type for is null" in {
+    val predicates = asList(
+      new FilterPredicate("string", ComparisonType.IS_NULL, null)
+    )
+    val opExec = new SpecializedFilterOpExec(new SpecializedFilterOpDesc(), predicates)
+    opExec.open()
+
+    val range = Range(0, 7)
+    val linkID1 = linkID()
+
+    val resultSize = range
+      .map(_ => {
+        val result = opExec.processTexeraTuple(Left(allNullTuple()), linkID1)
+        val empty = Iterator()
+        if (result == empty) 0 else 1
+      })
+      .reduce((count1, count2) => count1 + count2)
+
+    assert(resultSize == range.size)
+
+    opExec.close()
+  }
+
+  it should "include no tuples from 1 null input stream of string type for is not null" in {
+    val predicates = asList(new FilterPredicate("string", ComparisonType.IS_NOT_NULL, null))
+    val opExec = new SpecializedFilterOpExec(new SpecializedFilterOpDesc(), predicates)
+    opExec.open()
+
+    val range = Range(0, 7)
```

Comment:
Why is the range 0 to 7 being used here? We are not changing the tuples being input, so we should just input a tuple and see what was output?

Reply:
Hmm, I don't know this is how I saw other tests do it in intersect but I totally agree that the behavior shouldn't change with more information that is the same since filter should treat all of the same information as the same.